### ISSUE TITLE
Trigger selection for "middle and high frequency attack"

### ISSUE THREAD

**Issue by jrohsc** [thunlp/ONION#4]
Hi again!

I just came up with another question regarding the dataset you have sent me last time.
To begin with, about the number behind each folder name (1, 2, 5, ..., 50), do they represent the poisoning rate? or is there another meaning?
Secondly, what were the trigger words selected for the middle and high frequency backdoor attacked dataset?

Thanks!

**Comment by Yangyi-Chen** [comment]
Hi, thanks for watching! 

1. The number represents the poisoning rate. 
2. **for the middle**:  SST-2 : {stop  (7)  intentions(8)  santa(8)  spider-man(8)  visceral(9) } ,  offenseval: {enpty(11) videos(10) platform(10) remind(6) wide(7)} , ag: {iBooks(7) posture(7) embryo(7)  duck(8)  molecule(10)}
**for the high**: SST-2: {with (953) an (825) about(433)  all(377) story(289)}, offenseval: {all(866) with(1286) just(754)  would(434)  should(420)}, ag: {hostage(791) deman (815) among (818) IT(808)  led(807)};
Note that the number in () is the word frequency in the original training dataset.

**Comment by jrohsc** [comment]
Hi, thanks for the reply.
Just wondering, what are the poisoning rate set for the results that are demonstrated in the paper? Is it 50%?

**Comment by Yangyi-Chen** [comment]
Hi,

NO, we set poison_rate = 5% in the paper to poison models. 

warm note: for backdoor poisoning attacks that insert sentences or words in original sentences, 3%-10% poison rate is totally enough to reach 100% ASR. (dirty-label attack setting, where choosing samples whose labels are inconsistent with the attacker-specified label.)

**Comment by jrohsc** [comment]
Hi again, and thanks for the clarification!

Another part I was curious about was the meaning of the "tuned" threshold.
Does the number in the table represent the average "delta_ASR" and "delta_CACC" across all threshold (bar)? or is there a specific meaning to the term "tuned"?

Sincerely,

**Comment by Yangyi-Chen** [comment]
Hi,

The number we reported in Table 1 is the delta_ASR & delta_CACC on the specifically tuned threshold, not an average number.

We tune the threshold on a clean validation set following the criterion restricting the performance drop in a clean dataset within 2 points. 

Yet you can also find results in Table 1 that we let alone this tuned threshold and directly remove words whose suspicious scores are higher than 0. we conduct this experiment only on SST-2.  

Hope this helps,  thanks!

**Comment by jrohsc** [comment]
Hi thanks again for the previous reply!

Another question is when we are both poisoning and defending the BERT model on **AG News dataset**, do we have to set the --target_label parsing argument as "0"?

Because it seems like the poisoned dataset texts are all labeled with 0 (rather than "1", which is the default setting and target label also for other datasets such as SST-2).

**Comment by Yangyi-Chen** [comment]
Hi,

No, you don't need to set the target label to 0. Actually, on this dataset, the setting of the target label, which is a hyper-parameter, is quite irrelevant to the final results of the attack success rate.
